Create a "Supported Cluster Modifications" topic (#907)
astronomer/docs: astro/modify-cluster.md

Unless otherwise specified, new Clusters on Astro are created with a set of default configurations. Depending on your use case, you may decide that you want to modify an existing Cluster to run a different configuration.

For example, if you have a new set of DAGs that require significantly more CPU and Memory than your existing workloads, you might be interested in modifying a cluster on AWS to run `m5.8xlarge` nodes instead of `m5.4xlarge` nodes. You might also want to modify a cluster's maximum node count from the default of 20 to better fit your expected workload.

> ## Prerequisites

Reviewer: Looking at the header hierarchy, I actually wonder if it'd make sense to do:

## Review supported cluster modifications
- This is where you'd have the list you added
- Link to AWS + GCP resource reference

## Reach out to Astronomer

## Confirm your cluster change
- Let's actually replace the "confirm with Astronomer" piece to just have them look in the Cloud UI? We can have a statement that says Astronomer will confirm, but we should add that the user can also just go to Cloud UI:

![image](https://user-images.githubusercontent.com/31111752/179050346-770c0094-53fb-4dd0-89a4-9a36a26ae8de.png)

If you don't have a cluster on Astro, follow the instructions to [Install Astro on AWS](install-aws.md) or [GCP](install-gcp.md). If you have an existing cluster and are interested in creating additional clusters, read [Create a cluster](create-cluster.md).

## Supported cluster modifications

Some cluster and Deployment-level modifications can be completed only by Astronomer support. These include:

- [Creating a new cluster](create-cluster.md).
- Updating a cluster's worker instance type.
- Updating the maximum node count of an existing cluster.

> - [Creating a VPC connection](connect-external-services.md#vpc-peering) between a cluster and a target VPC.

Author: We have docs for it now approved by Deepak: https://github.com/astronomer/docs/pull/882#event-7015750790

- Running images from a private registry with the [KubernetesPodOperator](kubernetespodoperator#run-images-from-a-private-registry).

## Step 1: Submit a request to Astronomer

To modify an existing cluster in your Organization, first verify that the change you want to make is supported by reading the resource reference documentation for either [AWS](resource-reference-aws.md) or [GCP](resource-reference-gcp.md). Then, reach out to [Astronomer support](https://support.astronomer.io).

## Step 2: Confirm the modification

If the modification you requested is supported, Astronomer will notify you as soon as it's possible to complete the modification.

> Modifications to an existing cluster might take a few minutes to complete, but you can expect no downtime during the process. Astro is built to ensure a graceful rollover, which means that the Airflow and Cloud UIs will continue to be available and your Airflow tasks will not be affected.

Reviewer: Yes -- tweaking wording here per @collinmcnulty suggestion

If you don't have a cluster on Astro, follow the instructions to [Install Astro on AWS](install-aws.md) or [GCP](install-gcp.md). If you have an existing cluster and are interested in creating additional clusters, read [Create a cluster](create-cluster.md).

## Supported cluster modifications

> Some cluster and Deployment-level modifications can be completed only by Astronomer support. These include:

Reviewer: ```suggestion
Some cluster and Deployment-level modifications require Astronomer support and cannot be completed with the Cloud UI or CLI. These include requests to:
```

> - [Creating a new cluster](create-cluster.md).

Reviewer: ```suggestion
- [Create a new cluster](create-cluster.md).
```

> - Deleting a cluster.

Reviewer: ```suggestion
- Delete an existing cluster.
```

> - Updating a cluster's worker instance type. See cloud resource references ([AWS](resource-reference-aws.md#deployment-worker-size-limits), [GCP](resource-reference-gcp.md#deployment-worker-size-limits)).

Reviewer: ```suggestion
- Update a cluster's worker instance type. See cloud resource references ([AWS](resource-reference-aws.md#deployment-worker-size-limits), [GCP](resource-reference-gcp.md#deployment-worker-size-limits)).
```

> - Updating the maximum node count of an existing cluster.

Reviewer: ```suggestion
- Update the maximum node count of an existing cluster.
```

- [Creating a VPC connection](connect-external-services.md#vpc-peering) between a cluster and a target VPC.

> - Running images from a private registry with the [KubernetesPodOperator](kubernetespodoperator#run-images-from-a-private-registry).

Reviewer: ```suggestion
- Run Docker images from a private registry with the [KubernetesPodOperator](kubernetespodoperator#run-images-from-a-private-registry).
```

## Step 1: Submit a request to Astronomer

To modify an existing cluster in your Organization, first verify that the change you want to make is supported by reading the resource reference documentation for either [AWS](resource-reference-aws.md) or [GCP](resource-reference-gcp.md). Then, reach out to [Astronomer support](https://support.astronomer.io).

## Step 2: Confirm the modification

If the modification you requested is supported, Astronomer will notify you as soon as it's possible to complete the modification.

> Modifications to an existing cluster might take a few minutes to complete, but you can expect no downtime during the process. Astro is built to ensure a graceful rollover, which means that the Airflow and Cloud UIs will continue to be available and your Airflow tasks will not be affected.

Reviewer: ```suggestion
Most modifications to an existing cluster take only a few minutes to complete and do not incur downtime. In these cases, the Airflow UI and Cloud UI will continue to be available and your Airflow tasks are not interrupted.

For modifications that do incur downtime, such as changing your cluster's node instance type, Astronomer support will inform you of the expected impact and ask that you confirm if you want to proceed.
```

> Once our team validates that the cluster configuration you requested is supported, we will let you know as soon as we are able to perform the change.

Reviewer 2: ```suggestion
 You are notified when Astronomer support validates that the cluster configuration you requested is supported and can be implemented.
```
